src/opover.d

```diff
@@ -1959,14 +1961,31 @@ extern (C++) static Dsymbol inferApplyArgTypesX(Expression ethis, FuncDeclaratio
         }
         if (!inferApplyArgTypesY(tf, parameters, 1))
             m = MATCHnomatch;
+
         if (m > match)
         {
             fd_best = f;
             fd_ambig = null;
             match = m;
         }
-        else if (m == match)
+        else if (m == match && fd_best)
```

Comment:
What happens with the case `m == match && !fd_best` ? Seems to me it will now fail to match, whereas it did before.